fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -99,7 +99,27 @@
      */
     private static final String INTERNAL_COMMIT_HOOK_PREFIX = "@__";
     private static final String AFTER_COMMIT_HOOK_NAME = INTERNAL_COMMIT_HOOK_PREFIX + "afterCommit";
-    private static final int MAX_TR_ID_SIZE = 100;
+
+    /**
+     * The maximum size for a transaction ID in bytes when serialized as UTF-8. This value is used to determine
+     * whether the transaction ID provided in
+     * {@link FDBDatabase#openContext(Map, FDBStoreTimer, FDBDatabase.WeakReadSemantics, FDBTransactionPriority, String)}
+     * should be truncated or dropped. Note that Java {@link String}s are encoded using UTF-16, so using
+     * {@link String#length()} is insufficient to know if the transaction ID will be too large if it contains
```

Comment:
Maybe you could mention a sufficient way (`Utf8.encodedLength`) here?

Reply:
Sure, took a stab at that